File: src/main/java/application/history/history/presentation/interfaces/api/GetHistoryController.java
package application.history.history.presentation.interfaces.api;

import application.history.history.application.services.GetHistoryByDateService;
import application.history.history.domain.entities.History;
import org.springframework.format.annotation.DateTimeFormat;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.RestController;

import java.time.LocalDate;
import java.util.Date;
import java.util.Optional;

@RestController
public class GetHistoryController {
    private final GetHistoryByDateService service;

    public GetHistoryController(GetHistoryByDateService getHistoryService) {
        this.service = getHistoryService;
    }

    @GetMapping("/histories/{date}")
    public ResponseEntity<Optional<History>> getHistory(@PathVariable("date") @DateTimeFormat(pattern = "yyyy-MM-dd") LocalDate date) {
        return ResponseEntity.ok(this.service.run(date));
    }
}

File: src/main/kotlin/com/example/history/presentation/interfaces/api/GetHistoryController.kt
package com.example.history.presentation.interfaces.api

import com.example.history.application.services.FindHistoryUseCase
import com.example.history.domain.entities.History
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.PathVariable
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController

@RestController
@RequestMapping("/api/histories")
class GetHistoryController(private val service: FindHistoryUseCase) {

    @GetMapping("/{id}")
    fun getHistory(@PathVariable("id") id: String): ResponseEntity<History> {
        return ResponseEntity.ok(service.findById(id))
    }
}
